### DISCUSSION TITLE
[monorepo] What is the purpose of the "resolutions" key in this monorepo's package.json?

### DISCUSSION THREAD

**Discussion by silesky** [dotansimha/graphql-code-generator#8610] [Q&A] [answered]
Curious about why this is needed / what situations this is useful in? My understanding is that yarn workspaces automatically uses the simlinked versions of these, no matter where they live (as long as installed at the top level)?
```
"resolutions": {
    "prettier": "2.7.1",
    "babel-jest": "28.1.3",
    "jest-runner": "28.1.3",
    "graphql": "16.6.0",
    "graphql-language-service-interface": "2.10.2",
    "**/apollo-language-server/graphql": "^16.0.0",
    "**/@types/graphql-upload/graphql": "^16.0.0",
    "ts-node": "10.9.1"
  }
  ```
  
  I am running a monorepo with about the same stack, so its of particular interest.

**Comment by charlypoly** [comment, marked as answer]
Hi @silesky,

You can find the documentation for Yarn resolutions here: https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/

Please know that using it with `@graphql-codegen` packages won't work since it is used as a binary.
